addons/base/views.py

```diff
@@ -302,40 +302,27 @@ def get_authenticated_resource(resource_id):
     return resource
 
 
-def get_file_version_from_wb(waterbutler_data: dict) -> FileVersion:
-    path = waterbutler_data.get('path')
-    if waterbutler_data['provider'] != 'osfstorage' or not path:
-        return
-
-    path = waterbutler_data.get('path')
-    file_id = path.strip('/')
-    version = waterbutler_data.get('version')
-    if version:
-        try:
-            return FileVersion.objects.filter(
-                basefilenode___id=file_id,
-                identifier=int(version)
-            ).select_related('region').get()
-        except FileVersion.DoesNotExist:
-            raise HTTPError(http_status.HTTP_400_BAD_REQUEST)
-    else:
-        try:
-            return BaseFileNode.active.get(
-                _id=file_id,
-            ).versions.order_by(
-                '-created'
-            ).first()
-        except BaseFileNode.DoesNotExist:
-            raise HTTPError(http_status.HTTP_400_BAD_REQUEST)
+def _get_osfstorage_file_version(file_node: OsfStorageFileNode, version_string: str = None) -> FileVersion:
+    if not (file_node and file_node.is_file):
+        return None
+
+    version = int(version_string) if version_string else file_node.versions.count()
+    try:
+        return FileVersion.objects.select_related('region').get(
+            basefilenode=file_node,
+            identifier=version
```

Comment:
Minor: `identifier` is a `CharField`, and this is being cast to an `int` above. It _should_ get cast back to `str` somewhere in query planning or execution, though, so likely not a concern.

Reply:
Actually much nicer to cast an int to a string than a `str | None` to an int.